plugins/Referrers/API.php

```diff
@@ -328,22 +330,88 @@ public function getUrlsFromWebsiteId($idSite, $period, $date, $idSubtable, $segm
+            return Social::getInstance()->getMainUrlFromName($name);
+        }));
+
+        $dataTable = $this->completeSocialTablesWithOldReports($dataTable, $idSite, $period, $date, $segment, $expanded, $flat);
+
+        $dataTable->filter('MetadataCallbackAddMetadata', array('url', 'logo', function ($url) { return Social::getInstance()->getLogoFromUrl($url); }));
+
+        return $dataTable;
+    }
+
+    private function completeSocialTablesWithOldReports($dataTable, $idSite, $period, $date, $segment, $expanded, $flat)
+    {
+        return $this->combineDataTables($dataTable, function() use ($idSite, $period, $date, $segment, $expanded, $flat) {
+            $dataTableFiltered = Archive::createDataTableFromArchive(Archiver::WEBSITES_RECORD_NAME, $idSite, $period, $date, $segment, $expanded, false);
+
+            $this->filterWebsitesForSocials($dataTableFiltered, $idSite, $period, $date, $segment, $expanded, $flat);
+
+            return $dataTableFiltered;
+        });
+    }
+
+    protected function combineDataTables($dataTable, $callbackForAdditionalData)
+    {
+        $isMap = false;
+        $hasEmptyTable = false;
+        if ($dataTable instanceof DataTable\Map) {
+            $isMap = true;
+            $dataTables = $dataTable->getDataTables();
+        } else {
+            $dataTables = [$dataTable];
+        }
+
+        foreach ($dataTables as $table) {
+            if ($table instanceof DataTable && !$table->getRowsCountWithoutSummaryRow()) {
+                $hasEmptyTable = true;
+                break;
+            }
+        }
+
+        if ($hasEmptyTable) {
+
+            $dataTablesForCompletion = $callbackForAdditionalData();
+
+            if (!$isMap) {
+                $dataTable = $dataTablesForCompletion;
+            } else {
+                $filteredTables = $dataTablesForCompletion->getDataTables();
+                foreach ($dataTable as $label => $table) {
+                    if ($table instanceof DataTable && !$table->getRowsCountWithoutSummaryRow() && !empty($filteredTables[$label])) {
+                        $dataTable->addTable($filteredTables[$label], $label);
+                    }
+                }
+            }
+        }
+
+        return $dataTable;
+    }
 
-        $dataTable->filter('ColumnCallbackDeleteRow', array('label', function ($url) { return !Social::getInstance()->isSocialUrl($url); }));
-        $dataTable->filter('ColumnCallbackAddMetadata', array('label', 'url', function ($url) { return Social::getInstance()->getMainUrl($url); }));
-        $dataTable->filter('GroupBy', array('label', function ($url) { return Social::getInstance()->getSocialNetworkFromDomain($url); }));
+    /**
+     * @param DataTable $dataTable
+     */
+    protected function filterWebsitesForSocials($dataTable, $idSite, $period, $date, $segment, $expanded, $flat)
+    {
+        $dataTable->filter('ColumnCallbackDeleteRow', array('label', function ($url) {
```

Comment:
Might be good to check performance at some stage with a profiler on real data ideally just to see... especially re the groupby but also everything else like those filters.